Assets/Scripts/Battlefield/AbstractTurnBrain.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using SwordAndBored.Battlefield.CreaturScripts;
+using SwordAndBored.Battlefield.TurnMechanism;
+using SwordAndBored.Battlefield.CameraUtilities;
+
+
+namespace SwordAndBored.Battlefield
+{
+    public abstract class AbstractTurnBrain : MonoBehaviour
+    {
+        public TurnManager turnManager;
```

Comment:
Since you are a component of a unit, do not have access to the turnManager but just access the player you are attached to. This will cause every brain for every unit to run UpdateBrainHandler() but on the activePlayer from the turn manager every frame. 

The modularity of having a brain for each player is lost if you ask the TurnManager for the activePlayer. See if you can architect the code so that each brain's update happens only if the player is the activePlayer. 

My suggestion would be running unit.DoTurn() in the turnManagers update() function which in fact runs myBrain.UpdateBrainHandler() along with other things like the gui and such or you can have the currently active gameobject know it is the activePlayer and tell it to take control of gui and run its brain while it is active and the turn manager can make a unit active or unactive.